### PR TITLE
refactor: remove table ident

### PR DESCRIPTION
**Pull request by MichaelScofield**
I hereby agree to the terms of the [GreptimeDB CLA](https://gist.github.com/xtang/6378857777706e568c1949c7578592cc)

## What's changed and what's your intention?

This PR:
1. remove TableIdent, use TableId directly
2. use the latest greptime-proto
3. independently invalidate table id cache and table name cache

## Checklist

- [ ]  I have written the necessary rustdoc comments.
- [ ]  I have added the necessary unit tests and integration tests.

### CODE REVIEW

File: src/common/meta/src/cache_invalidator.rs [hunk 1]
```diff
@@ -14,19 +14,23 @@
...
     pub subject: Option<String>,
 }
 
 #[async_trait::async_trait]
 pub trait CacheInvalidator: Send + Sync {
     // Invalidates table cache
-    async fn invalidate_table(&self, ctx: &Context, table_ident: TableIdent) -> Result<()>;
+    async fn invalidate_table_id(&self, ctx: &Context, table_id: TableId) -> Result<()>;
+
+    async fn invalidate_table_name(&self, ctx: &Context, table_name: TableName) -> Result<()>;
```
Comment (WenyXu) [inline src/common/meta/src/cache_invalidator.rs:33]: Unfortunately, The `TableInfo` also contains the name attribute.

File: src/datanode/src/heartbeat/handler.rs [hunk 2]
```diff
@@ -46,7 +47,7 @@ impl RegionHeartbeatResponseHandler {
             Instruction::OpenRegion(region_ident) => {
                 let region_id = Self::region_ident_to_region_id(&region_ident);
                 let open_region_req = RegionRequest::Open(RegionOpenRequest {
-                    engine: region_ident.table_ident.engine,
+                    engine: default_engine().to_string(),
```
Comment (WenyXu) [inline src/datanode/src/heartbeat/handler.rs:50]: `RegionIdent` should contain the engine field.

File: src/frontend/src/catalog.rs [hunk 3]
```diff
@@ -65,13 +66,8 @@ pub struct FrontendCatalogManager {
...
 #[async_trait::async_trait]
 impl CacheInvalidator for FrontendCatalogManager {
-    async fn invalidate_table(&self, _ctx: &Context, table_ident: TableIdent) -> MetaResult<()> {
-        let table_id = table_ident.table_id;
-        let key = TableNameKey::new(
-            &table_ident.catalog,
-            &table_ident.schema,
-            &table_ident.table,
-        );
+    async fn invalidate_table_name(&self, _ctx: &Context, table_name: TableName) -> MetaResult<()> {
```
Comment (WenyXu) [inline src/frontend/src/catalog.rs:69]: I don't think it's a good idea to separate `invalidate_table` into `invalidate_table_name` and `invalidate_table_id`. It incurs cache inconsistency. Especially `TableInfo` and `TableName` should be invalidated at the same time.

**Comment by WenyXu** [comment]
Let's add some flag in the `InvalidateTable` instruction instead of adding another instruction.

 e.g., 
```
InvalidateTable {
    with_route: bool
} 
```
By default, `with_route` is true.

File: src/common/meta/src/cache_invalidator.rs (hunk 1 above)
Comment (MichaelScofield) [reply src/common/meta/src/cache_invalidator.rs:33]: I think the table name in `TableInfo` should be removed. It should simply contain table id. I will leave a todo for myself.

File: src/datanode/src/heartbeat/handler.rs (hunk 2 above)
Comment (MichaelScofield) [reply src/datanode/src/heartbeat/handler.rs:50]: @waynexia Can the field `engine` in `RegionOpenRequest` be removed? Don't see the actual usage.

File: src/frontend/src/catalog.rs (hunk 3 above)
Comment (MichaelScofield) [reply src/frontend/src/catalog.rs:69]: I think it won't cause any inconsistency if the `TableInfo` does not contain table name. In that case, the `TableInfo` and `TableName` are orthogonal, i.e. they are used separately independently.

Comment (WenyXu) [reply src/frontend/src/catalog.rs:69]: Trust me, any accidental misuse may lead to bugs 🐛 . Or we should change the `TableInfo::name`'s visibility to pub(crate) to protect potential misuse.

Comment (MichaelScofield) [reply src/frontend/src/catalog.rs:69]: I believe in you. However, the bugs are guarded by thoroughly tests. I will remove the catalog, schema and table names in `TableInfo`.

File: src/datanode/src/heartbeat/handler.rs (hunk 2 above)
Comment (fengjiachun) [reply src/datanode/src/heartbeat/handler.rs:50]: > @waynexia Can the field `engine` in `RegionOpenRequest` be removed? Don't see the actual usage.

Datanode needs to know which engine to use to open the region: https://github.com/GreptimeTeam/greptimedb/blob/develop/src/datanode/src/region_server.rs#L207

**Review by fengjiachun — APPROVED** [review]

**Review by WenyXu — APPROVED** [review]

**Pull request merged**